docs/pulsar_geo_replication.md

```diff
@@ -0,0 +1,185 @@
+
+
+## How to setup a Geo Replication
+
+This is a demo shows how to setup a geo replication.
+
+The demo will set up geo-replication between cluster1, and cluster2, the relations are shown in the diagram.
+```mermaid
+graph TD;
+   cluster1-->cluster2;
+   cluster2-->cluster1;
+```
+
+The demo will use two pulsar clusters in the different namespaces in the same k8s cluster.
+The cluster names are `cluster1-sn-platform` and `cluster2-sn-platform`.
+
+
+### Prerequisites
+1. You have two pulsar clusters at least.
+2. Make sure both clusters can access each other.
+
+### Update the existing PulsarConnection
+You need to add the extra fields `clusterName` and `brokerServiceURL` for the existing PulsarConnection.
+
+```yaml
+apiVersion: resource.streamnative.io/v1alpha1
+kind: PulsarConnection
+metadata:
+  name: cluster1-connection
+  namespace: cluster1
+spec:
+  clusterName: cluster1-sn-platform # the local cluster name
+  adminServiceURL: http://cluster1-sn-platform-broker.cluster1.svc.cluster.local:8080
+  brokerServiceURL: http://cluster1-sn-platform-broker.cluster1.svc.cluster.local:6650
+  authentication:
+    token:
+      secretRef:
+        name: cluster1-sn-platform-vault-secret-env-injection
+        key: brokerClientAuthenticationParameters
+```
+
+### Create a Destination PulsarConnection
+First, you need to create a destination PulsarConnection, this connection indicates the info of the remote pulsar cluster.
+You need to add two extra fields `clusterName` and `brokerServiceURL` for this PulsarConnection.
+
+Notice, the token of the authentication only support using `value` now.
+
+```yaml
+apiVersion: resource.streamnative.io/v1alpha1
+kind: PulsarConnection
+metadata:
+  name: cluster1-dest-connection
+  namespace: cluster1
+spec:
+  clusterName: cluster2-sn-platform # the remote cluster name
+  adminServiceURL: http://cluster2-sn-platform-broker.cluster2.svc.cluster.local:8080 # the remote pulsar admin service
+  brokerServiceURL: pulsar://cluster2-sn-platform-broker.cluster2.svc.cluster.local:6650 # the remote pulsar broker service
+  authentication:
+    token:
+      value: YzBkMzVjOGEtNWRiNi0yZGM3LTI4NjktZTM1NGU1YmJiNTY4OjAwZTk2MzgzLWFjY2YtMGMzOC1kNmRlLTY1ODc0ODkxNmE1ZQo=
+
+```
+
+
+### Create a PulsarGeoReplication
+It means setup the geo replication from the `cluster1` to `cluster2`. The operator will create a new cluster called `cluster2-sn-platform`.
+
+```yaml
+apiVersion: resource.streamnative.io/v1alpha1
+kind: PulsarGeoReplication
+metadata:
+  name: cluster1-geo
+  namespace: cluster1
+spec:
+  connectionRef:
+    name: cluster1-connection
+  destinationConnectionRef:
+    name: cluster1-dest-connection
+  lifecyclePolicy: CleanUpAfterDeletion
+```
+
+### Grant the permission for the tenant
+
+You can create a new tenant or update the existing the tenant with the field `geoReplicationRefs`. It will assign the `cluster2-sn-platform` to the tenant.
```

Comment:
```suggestion
You can create a new tenant or update an existing tenant by adding the field `geoReplicationRefs`. It will add the cluster `cluster2-sn-platform` to the tenant.
```